src/org/zaproxy/zap/extension/pscanrulesBeta/CookieLooselyScopedScanner.java

```diff
@@ -104,8 +104,8 @@ private boolean isLooselyScopedCookie(HttpCookie cookie, String host) {
 			return cookieDomain.equals(host);
 		}
 		
-		// otherwise, remove the '.' and compare the result with the host
-		if (cookieDomains.length != 2) {
+		// otherwise, remove the '.' (if cookie domain is not empty) and compare the result with the host
+		if (!cookieDomain.isEmpty() && cookieDomains.length != 2) {

```

Comment:
If `cookieDomain.isEmpty()` wouldn't the code have returned at L91 and not gotten here?